lib/active_record/virtual_attributes/virtual_arel.rb

```diff
@@ -14,6 +14,30 @@ class Arel::Nodes::Grouping
     end
 
     module VirtualArel
+      class ATProxy < Arel::Table
+        attr_accessor :klass
+
+        def [](name, table = self)
```

Comment:
Also is this code "our" code or code copied from Rails and then modified somehow?  If the latter, it might be useful to tack in a comment linking to where it came from and describing what was changed.  Then, in the git history, having a copy of the original with our changes on top in a separate commit so we can see the diff (I think this was kind of done already, but I figured I'd call out that out anyway)

Reply:
this is ours - actually copied from our original `arel_attribute` code
I'll add a comment defining what it does

Reply:
added comments to explain what is happening. hope this fixes the issues